src/ng/directive/form.js

```diff
@@ -228,16 +234,29 @@ function FormController(element, attrs, $scope, $animate) {
    * saving or resetting it.
    */
   form.$setPristine = function () {
-    $animate.removeClass(element, DIRTY_CLASS);
-    $animate.addClass(element, PRISTINE_CLASS);
+    $animate.setClass(element, PRISTINE_CLASS, DIRTY_CLASS + ' ' + SUBMITTED_CLASS);
     form.$dirty = false;
     form.$pristine = true;
+    form.$submitted = false;
     forEach(controls, function(control) {
       control.$setPristine();
     });
   };
-}
 
+  /**
+   * @ngdoc function
+   * @name ng.directive:form.FormController#$setSubmitted
+   * @methodOf ng.directive:form.FormController
+   *
+   * @description
+   * Sets the form to its submitted state.
+   */
+  form.$setSubmitted = function () {
+    $animate.addClass(element, SUBMITTED_CLASS);
+    form.$submitted = true;
+    parentForm.$setSubmitted();
```

Comment:
Yep, I have a multi-step form (aka wizard). I use .ng-submitted to style invalid inputs and have a plugin that detects if forms is dirty.

On pre-exit a state:

```js
nestedForm.$setSubmitted();
if (!nestedForm.$valid) return;
```

On entering a sub-state / nested form:

```js
this.target.mainForm.$setPristine();
```

Style:

```css
.ng-submitted [ng-model].ng-invalid {
    border: 1px solid red;
}
```

The border keeps being draw because there's a form styled with .ng-submitted further up. And I actually use $setPristine() because it clears $submitted, but then it breaks form dirty-checking plugin.

Some strategies that could it (if they existed):

* form.$setSubmitted(state: boolean);
* form.$setSubmitted(updateParents: boolean);

I'm not sure yet about the invariants that could break with that. WDYT?